src/pudl/extract/epacems.py

```diff
@@ -172,71 +121,40 @@ def __init__(self, datastore: Datastore):
         """Construct datastore wrapper for loading raw EPA CEMS data into dataframes."""
         self.datastore = datastore
 
-    def get_data_frame(self, partition: EpaCemsPartition) -> pd.DataFrame:
+    def get_data_frame(self, partition: EpaCemsPartition) -> pl.LazyFrame:
         """Constructs dataframe from a zipfile for a given (year_quarter) partition."""
         with (
             self.datastore.get_zipfile_resource(
                 "epacems", **partition.get_filters()
             ) as zf,
             zf.open(str(partition.get_quarterly_file()), "r") as csv_file,
         ):
-            df = self._csv_to_dataframe(
-                csv_file,
-                ignore_cols=API_IGNORE_COLS,
-                rename_dict=API_RENAME_DICT,
-                dtype_dict=API_DTYPE_DICT,
+            lf = pl.scan_csv(csv_file, low_memory=True, schema_overrides=API_DTYPE_DICT)
+            lf = (
+                lf.drop(
+                    list(set(lf.collect_schema().names()) - set(API_RENAME_DICT.keys()))
+                )
```

Comment:
Would a `.select(API_RENAME_DICT.keys())` here not do the same thing without having to collect the schema first?